granulate_utils/metrics/sampler.py

```diff
@@ -307,7 +309,7 @@ def discover(self) -> bool:
                 )
                 have_conf = True
 
-        if have_conf and self._spark_samplers == []:
+        if have_conf and self._collectors == []:
```

Comment:
This makes no sense - you added an `assert` for it in line 286. What you probably need to assert for is that `_init_collectors` was not called, so you can keep the `collectors == []` assert, and remove the check here.

Reply:
forgot to commit that